nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/GetAzureCosmosDBRecord.java

```diff
@@ -0,0 +1,266 @@
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.schema.access.SchemaNotFoundException;
+import org.apache.nifi.serialization.RecordSetWriter;
+import org.apache.nifi.serialization.RecordSetWriterFactory;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordSchema;
+
+
+@Tags({ "azure", "cosmos", "record", "read", "fetch" })
+@InputRequirement(Requirement.INPUT_ALLOWED)
+@CapabilityDescription("A record-oriented GET processor that uses the record writers to write the Azure Cosmos SQL select query result set.")
+public class GetAzureCosmosDBRecord extends AbstractAzureCosmosDBProcessor {
+    public static final PropertyDescriptor WRITER_FACTORY = new PropertyDescriptor.Builder()
+        .name("record-writer-factory")
+        .displayName("Record Writer")
+        .description("The record writer to use to write the result sets")
+        .identifiesControllerService(RecordSetWriterFactory.class)
+        .required(true)
+        .build();
+    public static final PropertyDescriptor SCHEMA_NAME = new PropertyDescriptor.Builder()
+        .name("schema-name")
+        .displayName("Schema Name")
+        .description("The name of the schema in the configured schema registry to use for the query results")
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .defaultValue("${schema.name}")
+        .required(true)
+        .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+        .name("sql-core-document-query")
+        .displayName("SQL Core Document Query")
+        .description("The SQL select query to execute. "
+                + "This should be a valid SQL select query to Cosmos DB with core sql api")
+        .required(true)
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    public static final PropertyDescriptor MAX_RESPONSE_PAGE_SIZE = new PropertyDescriptor.Builder()
+        .name("max-page-size")
+        .displayName("Max Page Size")
+        .description("The maximum number of elements in a response page from Cosmos DB")
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+        .defaultValue("10")
+        .build();
+
+    private final static Set<Relationship> relationships;
+    private final static List<PropertyDescriptor> propertyDescriptors;
+    private ComponentLog logger;
+    private final static ObjectMapper mapper;
```

Comment:
Hi, would you please reorder the qualifiers to "public static final" to comply with JSL :) ?

Reply:
You mean private right?

Reply:
Yes

Reply:
Thank you for reviewing my PR. I have taken most of suggestions.

Reply:
Thank you for the improvements!